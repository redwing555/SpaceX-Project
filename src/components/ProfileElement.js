import { useSelector } from 'react-redux';
import '../styles/ProfileElement.css';

const ProfileElement = () => {
  const missionsData = useSelector((state) => state.missionsReducer);
  const rockets = useSelector((state) => state.rocketsReducer.rockets);
  let reservedRockets;
  if (rockets) {
    reservedRockets = rockets.filter((rocket) => rocket.reserved);
  }

  return (

    <div className="profile-container">
      <div className="profile">

        <h3 className="reservations">Rockets reservations</h3>
        {reservedRockets && (
        <ul className="profile-list">
          {reservedRockets.map((rocket) => (
            <li className="profile-name" key={rocket.id}>
              {rocket.name}
            </li>
          ))}

        </ul>
        )}
      </div>
      <div className="profile">
        <h3 className="reservations">Missions reservations</h3>
        <ul className="profile-list">
          {missionsData.filter((data) => data.reserved)
            .map((item) => (<li className="profile-name" key={item.mission_id}>{item.mission_name}</li>))}
        </ul>
      </div>
    </div>
  );
};

export default ProfileElement;
