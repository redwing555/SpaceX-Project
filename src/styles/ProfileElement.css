.profile-container{
    display: flex;
    gap: 5rem;
}



.reservations{
    font-family: 'Red Hat Mono', monospace;
    font-size: 2rem;
  font-weight: bold;
  padding-top: 0.5rem;
  margin-left: 3rem;
  padding-bottom: 2rem;
  
}


.profile-name{
    font-family: sans-serif;
    font-size: 1.5rem;
    text-align: center;
}

.profile-list{
    display: flex;
    gap: 0.5rem;
    border: 1px solid gray;
    margin-right: 3rem;
    margin-left: 2rem;
}

.profile-list li{
    border-bottom: 0.5px solid gray;
    text-align: center;
    margin-right: 2rem;
    padding: 1rem;

}