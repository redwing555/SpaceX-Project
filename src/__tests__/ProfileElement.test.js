import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ProfileElement from '../components/ProfileElement';

describe('Navigation Links Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <ProfileElement />
      </Provider>,
    );
    expect(screen.queryByText(/Missions reservations/)).toBeTruthy();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProfileElement />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
