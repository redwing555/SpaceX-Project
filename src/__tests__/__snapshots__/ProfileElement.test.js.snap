// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Links Tests renders correctly 1`] = `
<div
  className="profile-container"
>
  <div
    className="profile"
  >
    <h3
      className="reservations"
    >
      Rockets reservations
    </h3>
  </div>
  <div
    className="profile"
  >
    <h3
      className="reservations"
    >
      Missions reservations
    </h3>
    <ul
      className="profile-list"
    />
  </div>
</div>
`;
